src/zrc/sync/signals.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+from django.urls import reverse
+
+import requests
+from vng_api_common.models import APICredential
+from vng_api_common.utils import get_uuid_from_path
+from zds_client import Client, extract_params, get_operation_url
+
+from zrc.datamodel.models import ZaakInformatieObject
+
+logger = logging.getLogger(__name__)
+
+
+class SyncError(Exception):
+    pass
+
+
+def sync_create(relation: ZaakInformatieObject):
+    operation = 'create'
+
+    # build the URL of the Zaak
+    path = reverse('zaak-detail', kwargs={
+        'version': settings.REST_FRAMEWORK['DEFAULT_VERSION'],
+        'uuid': relation.zaak.uuid,
+    })
+    domain = Site.objects.get_current().domain
+    protocol = 'https' if settings.IS_HTTPS else 'http'
+    zaak_url = f'{protocol}://{domain}{path}'
+
+    logger.info("Zaak: %s", zaak_url)
+    logger.info("Informatieobject: %s", relation.informatieobject)
+
+    # Define the remote resource with which we need to interact
+    resource = 'objectinformatieobject'
+    client = Client.from_url(relation.informatieobject)
+
+    # TODO?
+    client.auth = APICredential.get_auth(relation.informatieobject)
+
+    try:
+        operation_function = getattr(client, operation)
+        operation_function(
+            resource,
+            {'object': zaak_url, 'informatieobject': relation.informatieobject, 'objectType': 'zaak'}
+        )
+    except Exception as exc:
+        logger.error(f"Could not {operation} remote relation", exc_info=1)
+        raise SyncError(f"Could not {operation} remote relation") from exc
+
+
+def sync_delete(relation: ZaakInformatieObject):
+    operation = 'delete'
+
+    # build the URL of the Zaak
+    path = reverse('zaak-detail', kwargs={
+        'version': settings.REST_FRAMEWORK['DEFAULT_VERSION'],
+        'uuid': relation.zaak.uuid,
+    })
+    domain = Site.objects.get_current().domain
+    protocol = 'https' if settings.IS_HTTPS else 'http'
+    zaak_url = f'{protocol}://{domain}{path}'
+
+    logger.info("Zaak: %s", zaak_url)
+    logger.info("Informatieobject: %s", relation.informatieobject)
+
+    # Define the remote resource with which we need to interact
+    resource = 'objectinformatieobject'
+    client = Client.from_url(relation.informatieobject)
+    client.auth = APICredential.get_auth(relation.informatieobject)
+
+    # Retrieve the url of the relation between the object and the
+    response = client.list(resource, query_params={'object': zaak_url})
```

Comment:
Needs to include `'informatieobject': relation.informatieobject` as well, since those two fields make up the unique entry. You may easily get multiple objects in the response if multiple documents are linked to the zaak.